lib/fastlane/actions/upload_artifact.rb

```diff
@@ -0,0 +1,114 @@
+module Fastlane
+  module Actions
+    class UploadArtifactAction < Action
+      def self.run(params)
+        require 'artifactory'
+        file_path = File.absolute_path(params[:file])
+        if File.exist? file_path
+          client = connect_to_artifactory(params)
+          artifact = Artifactory::Resource::Artifact.new
+          artifact.client = client
+          artifact.local_path = file_path
+          artifact.checksums = {
+              "sha1" => Digest::SHA1.file(file_path),
+              "md5" => Digest::MD5.file(file_path)
+          }
+          Helper.log.info "Uploading file: #{artifact.local_path} ..."
+          upload = artifact.upload(params[:repo], params[:repo_path], params[:properties])
+          Helper.log.info "Uploaded Artifact:"
+          Helper.log.info "Repo: #{upload.repo}"
+          Helper.log.info "URI: #{upload.uri}"
+          Helper.log.info "Size: #{upload.size}"
+          Helper.log.info "SHA1: #{upload.sha1}"
+        else
+          Helper.log.info "File not found: '#{file_path}'"
+        end
+      end
+
+      def self.connect_to_artifactory(params)
+        keys = [:endpoint, :username, :password, :ssl_pem_file, :ssl_verify, :proxy_username, :proxy_password, :proxy_address, :proxy_port]
+        keys.each do |key|
+          config[key] = params[key] if params[key]
+        end
+        Artifactory::Client.new(config)
+      end
+
+      def self.description
+        'This action uploads an artifact to artifactory'
+      end
+
+      def self.is_supported?(platform)
+        true
+      end
+
+      def self.author
+        ["Joel Koglin"]
```

Comment:
Please use your username instead of your full name. This will later be used to link to your GitHub/Twitter profile.